py/objarray.c

```diff
@@ -405,12 +415,17 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
                     mp_seq_replace_slice_grow_inplace(o->items, o->len,
                         slice.start, slice.stop, src_items, src_len, len_adj, item_sz);
                 } else {
-                    mp_seq_replace_slice_no_grow(o->items, o->len,
-                        slice.start, slice.stop, src_items, src_len, item_sz);
-                    // Clear "freed" elements at the end of list
-                    // TODO: This is actually only needed for typecode=='O'
-                    mp_seq_clear(o->items, o->len + len_adj, o->len, item_sz);
-                    // TODO: alloc policy after shrinking
+                    if (o->base.type == &mp_type_memoryview) {
+                        // destination is memoryview use memcpy() instead of mp_seq_replace_slice_no_grow() below
+                        memcpy((uint8_t*)o->items + o->free + slice.start, src_items, src_len);
```

Comment:
So, this comment states the obvious. Instead, it should tell why everything uses mp_seq_replace_slice_no_grow(), and memoryview suddenly doesn't.